### DISCUSSION TITLE
Code for an online tool

### DISCUSSION THREAD

**Discussion by annieton** [community/community#41219] [New to GitHub]
### Select Topic Area

Question

### Body

Hello, I'm looking for code or script for the magic 8 ball online tool which can be installed on a WordPress blog like this website [https://magic8ballguide.com/](https://magic8ballguide.com/)
Screenshot below
![Capture](https://user-images.githubusercontent.com/120217780/206786637-572902e7-461c-4dda-9e54-12b79b660985.PNG)

**Comment by ghostinhershell** [comment]
Hey there!  👋 

Thanks for posting in the GitHub Community and welcome, @annieton! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. The Accessibility category is a place for our community to discuss and provide feedback on the digital accessibility of GitHub products. Digital accessibility means that GitHub tools, and technologies, are designed and developed so that people with disabilities can use them.

I've gone ahead and moved this to the correct category for you. Good luck!

**Comment by sheztechpk** [comment]
If you're building code for an online tool, simple utilities like [mytextrepeater.com](http://mytextrepeater.com/) can be a useful example. A text repeater shows how a small browser based tool can turn user input into practical output with minimal code. It's also a handy utility for testing repeated strings, formatting, and text generation.

**Comment by Lxcardoza993** [comment]
Here's a complete, self-contained Magic 8 Ball in a single HTML file — no frameworks, no account, works offline, free to use/modify. Swap the 20 phrases for any list you like.

```html
<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Magic 8 Ball</title>
<style>
  body { font-family: system-ui, sans-serif; background:#0e0e1a; color:#eee; display:grid; place-items:center; min-height:100vh; margin:0; }
  .ball { width:280px; height:280px; border-radius:50%; background: radial-gradient(circle at 35% 30%, #3a3a5c, #05050c 70%); display:grid; place-items:center; cursor:pointer; box-shadow: 0 12px 40px rgba(0,0,0,.6), inset 0 -10px 30px rgba(0,0,0,.55); }
  .face { width:120px; height:120px; background:#0b0b1f; border:2px solid #54547e; border-radius:12px; display:grid; place-items:center; text-align:center; padding:12px; }
  #answer { color:#9ecbff; font-size:15px; line-height:1.35; transition:opacity .15s; }
  .ball.shake { animation: shake .7s; }
  @keyframes shake { 0%,100%{transform:rotate(0)} 20%{transform:rotate(-8deg)} 40%{transform:rotate(8deg)} 60%{transform:rotate(-6deg)} 80%{transform:rotate(6deg)} }
  p.hint { color:#888; font-size:13px; text-align:center; }
</style>
</head>
<body>
<div>
  <div class="ball" id="ball" role="button" tabindex="0" aria-label="Ask the Magic 8 Ball">
    <div class="face"><span id="answer">Ask a question, then click</span></div>
  </div>
  <p class="hint">Click the ball · press <b>?</b> to ask</p>
</div>
<script>
  const answers = [
    "It is certain", "It is decidedly so", "Without a doubt", "Yes — definitely",
    "You may rely on it", "As I see it, yes", "Most likely", "Outlook good",
    "Yes", "Signs point to yes", "Reply hazy, try again", "Ask again later",
    "Better not tell you now", "Cannot predict now", "Concentrate and ask again",
    "Don't count on it", "My reply is no", "My sources say no",
    "Outlook not so good", "Very doubtful"
  ];
  const ball = document.getElementById('ball');
  const ans = document.getElementById('answer');
  function shake() {
    ball.classList.remove('shake'); void ball.offsetWidth; ball.classList.add('shake');
    ans.style.opacity = 0;
    setTimeout(() => {
      ans.textContent = answers[Math.floor(Math.random() * answers.length)];
      ans.style.opacity = 1;
    }, 350);
  }
  ball.addEventListener('click', shake);
  document.addEventListener('keydown', e => { if (e.key === '?') shake(); });
</script>
</body>
</html>
```

For a WordPress blog, three options with no plugin needed:

1. **Custom HTML block** — in the block editor, add a "Custom HTML" block and paste the full contents (switch to the Code Editor if the visual editor reformats it)
2. **Classic editor** — paste into the "Text" tab of a post or page
3. **Theme file** — save it as `magic8ball.html` (or a page template) in a child theme and link to it

If you want a typed-question input inside the circle or a share-the-answer button, say the word and I'll extend it. If this solves it, marking the answer as accepted would help the next person looking for the same tool. 🔮
